TS-4475: Log Collation Client SM, added VC_EVENT_INACTIVITY_TIMEOUT event handling.

pbchou opened the pull request:
Previously, this event was un-handled and would cause a core dump when it happened. This event is now treated as an additional error condition.

zwoop commented:
@oknet Can you review this too please?

zwoop commented:
[approve ci]

atsci commented:
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/491/ for details.

atsci commented:
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/388/ for details.

shinrich commented on proxy/logging/LogCollationClientSM.cc:
```diff
@@ -266,6 +266,9 @@ LogCollationClientSM::client_auth(int event, VIO * /* vio ATS_UNUSED */)
 
+  case VC_EVENT_INACTIVITY_TIMEOUT:
+    Debug("log-coll", "[%d]client::client_auth - Ignoring VC_EVENT_INACTIVITY_TIMEOUT", m_id);
+    return EVENT_CONT;
```
Is there a reason that you don't do the actions of the other terminal cases (EOS and ERROR)?

Also, should you through VC_EVENT_ACTIVE_TIMEOUT in here while you are at it?

pbchou commented:
@shinrich Thanks for your comments. I have been working with Oknet Xu on the Jira thread regarding this. I will combine the recommendations. Oknet also recommended to set a custom inactivity time-out for the net-vc for this.

pbchou commented:
Hi all. I updated the PR. The fix now does (1) handles both active time-out and inactivity time-out, (2) treats time-outs the same as EOS or ERROR events to handle any clean-ups, and (3) explicitly assigns the net-vc to have a time-out of 86400s so that it is not assigned the time-out defined in records.config by inactivity-cop. I would like to request that this be applied to both the master and 6.2.x branches.

shinrich commented:
Looks good to me.  I assume you set the very high timeout because the net connections for the log collation client aren't really like normal connections so the likely lower timeouts shouldn't apply.  Is that correct?

pbchou commented:
@shinrich @oknet -- Please see the updated PR. I added the same event handling to the host side. I also added two records.config options to make the time-out configurable on the host and client sides respectively. The default time-out of 86400s just corresponds to the default proxy.config.net.default_inactivity_timeout with a default configuration file.

pbchou commented:
@shinrich @oknet -- Hi. Did you get a chance to look at the latest changes?

oknet commented on doc/admin-guide/files/records.config.en.rst:
```diff
@@ -2703,6 +2703,14 @@ Logging Configuration
+.. ts:cv:: CONFIG proxy.config.log.collation_client_timeout INT 86400
+
+   The number of seconds before active or inactivity time-out events for the client side.
+
```
@pbchou Could you please add a comment to describe the reason of different default value for host and client ?

oknet commented:
@zwoop @pbchou Looks good to me!

pbchou commented:
@zwoop @oknet @shinrich -- I updated the documentation per Oknet's request. If everyone is good with this, then just a reminder that we also request that this should be back-ported 6.2.x as well since it fixes a crash bug.

oknet commented:
backport +1

pbchou commented:
@shinrich @oknet @zwoop -- Susan, are you the one assigned to do the merge? I think Oknet is okay with this, but I don't think we have your final thumbs up yet.

shinrich commented:
Sorry for the delay.  Looks good to me as well.
